util/llbutil/copyop.go

```diff
@@ -57,6 +58,37 @@ func CopyOp(srcState pllb.State, srcs []string, destState pllb.State, dest strin
 	return destState.File(fa, opts...)
 }
 
+// DockerfileCopy copies from `src` to `dest` using Docker's own custom image
+// for running the COPY command in Dockerfiles. This function is similar to
+// llb.Copy() but can accept RunOptions (for example, which may contain a mount).
+func DockerfileCopy(srcState pllb.State, src, dest string, opts []llb.RunOption, platform *specs.Platform) pllb.State {
```

Comment:
I would call this... `CopyViaRun` or something along those lines. The fact that the inspiration comes from an old Dockerfile-specific implementation is not particularly relevant.

Reply:
I called it `CopyWithRunOptions`